include/pagmo/population.hpp

```diff
@@ -411,6 +432,23 @@ class population
     }
 
 private:
+    // Short routine tp update the champion. Does nothing if the problem is MO
+    inline void update_champion(const vector_double& x, const vector_double& f)
+    {
```

Comment:
This function is not exception safe, as the copy assignment of ``f`` could throw after ``x`` has been set. A possible solution is to take the input params ``x`` and ``f`` by copy instead of const ref, and then move-assign them to ``m_champion_x`` and ``m_champion_f``.